src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -704,13 +711,7 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                 hasErrors = true;
                 Error(diagnostics, ErrorCode.ERR_TupleExplicitNamesOnAllMembersOrNone, node);
             }
-            var elements = elementTypes.ToImmutableAndFree();
-            if (elements.Length < 2 || elements.Length > 7)
-            {
-                // PROTOTYPE
-                diagnostics.Add(ErrorCode.ERR_PrototypeNotYetImplemented, node.Location);
-                return BadExpression(node);
-            }
+            ImmutableArray<TypeSymbol> elements = elementTypes.ToImmutableAndFree();
```

Comment:
No change needed. But we do sort of get lazy and use ```var``` with widely used APIs that "everybody" knows what they return. I guess ToImmutableAndFree is one of them.